- Description: dasd: I/O errors during DASD device qualification.
  Symptom:     ERP unsuccessful message followed by I/O error.
  Problem:     erp_requests are not recovered because the
               "DASD_CQR_FLAGS_USE_ERP" is not set by default.
  Solution:    Set the flag for erp_requests by default.
  Problem-ID:  -
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/block/dasd_erp.c	2 Jun 2003 14:24:37 -0000	1.10
+++ linux-2.5/drivers/s390/block/dasd_erp.c	6 May 2004 15:04:53 -0000	1.10.8.1
@@ -7,7 +7,7 @@
  * Bugreports.to..: <Linux390@de.ibm.com>
  * (C) IBM Corporation, IBM Deutschland Entwicklung GmbH, 1999-2001
  *
- * $Revision: 1.10 $
+ * $Revision: 1.10.8.1 $
  */
 
 #include <linux/config.h>
@@ -67,7 +67,8 @@
 	}
 	strncpy((char *) &cqr->magic, magic, 4);
 	ASCEBC((char *) &cqr->magic, 4);
-	atomic_inc(&device->ref_count);
+	set_bit(DASD_CQR_FLAGS_USE_ERP, &cqr->flags);
+	dasd_get_device(device);
 	return cqr;
 }
 

